app/controllers/membership_applications_controller.rb

```diff
@@ -47,11 +58,29 @@ def membership_application_params
     params.require(:membership_application).permit(*policy(@membership_application || MembershipApplication).permitted_attributes)
   end
 
+
   def get_membership_application
     @membership_application = MembershipApplication.find(params[:id])
   end
 
+
   def authorize_membership_application
     authorize @membership_application
   end
+
+
+  def new_upload_file(upload_file_param)
+    if upload_file_param['actual_files']
+      upload_file_param['actual_files'].each do |upload_file|
+
+        @uploaded_file = @membership_application.uploaded_files.create(actual_file: upload_file)
+        if @uploaded_file.valid?
+          flash[:notice] = "The file was uploaded: #{@uploaded_file.actual_file_file_name}"
+        else
+          flash[:error] = @uploaded_file.errors.messages
+        end
+      end
+
+    end
+  end
 end
```

Comment:
I guess that depends on project philosophy.  Some projects might require that a "slice" of functionality (such us upload membership files) include *all* elements of the complete stack - including unit, functional and acceptance tests.  Another philosophy is to break up such a "slice" into multiple stories, for different aspects of the stack.  

Either way works.  

The first approach does lead to bigger PR's - and this can be a negative if the architecture (and even the user story) has some ambiguity - which means smaller PR's are safer since less work goes into something that might change significantly with PR reviews.

Any other thoughts out there?